### PR TITLE
fix(AsyncStorageModule): Eliminate size constraint for async local storage

### LINKED ISSUE #282: AsyncStorageModule has value size limit
The F8 app is hitting a hard limit on the size of values stored in local settings.  We need to switch to something like SQLite to overcome this limitation.

### PR DESCRIPTION
**Pull request by rozele**
The app LocalSettings were not a good option for storing data, as they cap out at a relatively small storage capacity.

Switch to using the local file system as the key value store. Later, we can consider options like SQLite.

Fixes #282

### CODE REVIEW

**Comment by rozele** [comment]
@ebragge please review.

File: ReactWindows/ReactNative/Modules/Storage/AsyncStorageErrorHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            return GetError(key, "Invalid key");
+        }
+
+        public static JObject GetInvalidValueError(string key)
+        {
+            return GetError(key, "Invalid Value");
+        }
+
+        public static JObject GetDatabaseError(string key)
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Storage/AsyncStorageErrorHelpers.cs]: Delete, not used.

**Comment by ebragge** [comment]
Windows 10 IoT Core has a limitation of max 10000 files per folder.

The size of pair `{ "key_N" , "value_N" } `is about 120 bytes in the settings database.
Isn't minimum file size 4 KB?

Is the performance hit acceptable?

**Execution time:**
Surface Pro 3             

- 10000 items ..... 1:13 .....vs..... 0:08
- 20000 items ..... 2:29 .....vs..... 0:23                                
- 30000 items ..... 3:44 .....vs..... 0:45
- 40000 items ..... 4:59 .....vs..... 1:28 

```
        [TestMethod]
        public void AsyncStorageModule_Large()
        {
            var module = new AsyncStorageModule();
            var waitHandle = new AutoResetEvent(false);

            var error = default(JObject);
            var result = default(JArray);
            var callback = new MockCallback(res =>
            {
                error = res.Length > 0 ? (JObject)res[0] : null;
                result = res.Length > 1 ? (JArray)res[1] : null;
                waitHandle.Set();
            });

            var keyValues = new List<string[]>();
            for (var i=0; i<20000; i++)
            {
                var key = "key" + i;
                var value = "value" + i;
                keyValues.Add(new[]
                {
                    key,
                    value,
                });
            }

            module.multiSet(keyValues.ToArray(), callback);
            Assert.IsTrue(waitHandle.WaitOne());
            Assert.IsNull(error);
            Assert.IsNull(result);

            module.multiGet(new string[] { "key5000", }, callback);
            Assert.IsTrue(waitHandle.WaitOne());
            Assert.IsNull(error);
            Assert.AreEqual(result.Count, 1);
            Assert.AreEqual((result[0]).Last.Value<string>(), "value5000");
        }
```

**Comment by rozele** [comment]
We'll accept the performance hit for now, and revisit later if it's an issue for apps with many keys.

**Pull request merged**